### PR TITLE
Restore DNX_BUILD_KEY_FILE behavior (#3245).

### PR DESCRIPTION
**Pull request by cesarbs**
Fix https://github.com/aspnet/dnx/issues/3245

cc @anurse @davidfowl @emgarten

### CODE REVIEW

File: src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs [hunk 1]
```diff
@@ -128,6 +127,16 @@ private static CSharpCompilationOptions AddSigningOptions(CSharpCompilationOptio
...
         }
 
+        private static string GetKeyFileFullPath(string projectDirectory, string keyFile)
+        {
+            if (keyFile == null)
+            {
+                return null;
+            }
+
+            return Path.GetFullPath(Path.Combine(projectDirectory, keyFile));
```
Comment (moozzyk) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: int: consider ternary

File: src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs [hunk 2]
```diff
@@ -128,6 +127,16 @@ private static CSharpCompilationOptions AddSigningOptions(CSharpCompilationOptio
             return useOssSigning ? options.WithCryptoPublicKey(StrongNameKey) : options;
         }
 
+        private static string GetKeyFileFullPath(string projectDirectory, string keyFile)
+        {
+            if (keyFile == null)
```
Comment (moozzyk) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: I wonder if we shouldn't do the same if the keyFile is an empty string..

Comment (anurse) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: Yeah, `string.IsNullOrEmpty` here.

**Comment by anurse** [comment]
Minor comments. :shipit: once resolved

**Comment by muratg** [comment]
@emgarten do you need this in DNU?  Otherwise let's close this PR.

**Comment by cesarbs** [comment]
@anurse @moozzyk Went with `string.IsNullOrWhitespace` instead of `string.IsNullOrEmpty`.

**Comment by moozzyk** [comment]
That's fine

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**